### PR TITLE
FF7: Fix character naming screen not working correctly when Steam Input is enabled

### PR DESCRIPTION
**Pull request by maciej-trebacz**
## Summary

Fixes issue #622 by patching out functions that enable direct keyboard input on character naming screens. This is what Aali driver originally did and this change apparently wasn't ported to FFNx. See linked issue for more details.

### ACKs

- [x] I have updated the [Changelog.md](https://github.com/julianxhokaxhiu/FFNx/blob/master/Changelog.md) file
- [x] I did test my code on FF7
- [ ] I did test my code on FF8

### CODE REVIEW

File: src/ff7_opengl.cpp [hunk 1]
```diff
@@ -77,6 +77,12 @@ void ff7_init_hooks(struct game_obj *_game_object)
 	// Load Models atoi function
 	replace_call_function(ff7_externals.field_load_models_atoi, ff7_field_load_models_atoi);
 
+  // Restore Steam release behavior on character name screen when using gamepads in Steam Input mode
+  // Aali driver used to patch out these three functions to fix this issue
+  replace_function(ff7_externals.set_default_input_settings_save, noop);
+  replace_function(ff7_externals.keyboard_name_input, noop);
+  replace_function(ff7_externals.restore_input_settings, noop);
```
Comment (julianxhokaxhiu) [inline src/ff7_opengl.cpp:84]: Having those here will impact 1998 release as well. Better to move it where we do check if the game is the steam edition.

**Review by julianxhokaxhiu — CHANGES_REQUESTED** [review]
Thanks for the PR. I'm not against it but the way it is currently shape may create an issue with the non-Steam editions of the game. Left a comment on the impacted block.

File: src/ff7_opengl.cpp (hunk 1 above)
Comment (maciej-trebacz) [reply src/ff7_opengl.cpp]: @julianxhokaxhiu fixed.

**Review by julianxhokaxhiu — APPROVED** [review]

**Comment by julianxhokaxhiu** [comment]
Thank you!

**Pull request merged**